### PR TITLE
delete old log file if it exceeds kMaxLogFileSize of 25MiB

### PR DESCRIPTION
**Pull request by christopherfujino**
Addresses https://github.com/flutter/flutter/issues/150137

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/blob/main/docs/External-Package-Maintenance.md#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/unified_analytics/lib/src/log_handler.dart [hunk 1]
```diff
@@ -272,10 +272,18 @@ class LogHandler {
...
     try {
+      final stat = logFile.statSync();
+      List<String> records;
+      if (stat.size > kMaxLogFileSize) {
+        logFile.deleteSync();
+        logFile.createSync();
+        records = [];
+      } else {
+        records = logFile.readAsLinesSync();
+      }
```
Comment (andrewkolos) [inline pkgs/unified_analytics/lib/src/log_handler.dart:284]: For simplicity, consider
```suggestion
      final stat = logFile.statSync();
      if (stat.size > kMaxLogFileSize) {
        logFile.deleteSync();
        logFile.createSync();
      }
      List<String> records = logFile.readAsLinesSync();
```

File: pkgs/unified_analytics/test/log_handler_test.dart [hunk 2]
```diff
@@ -290,3 +334,51 @@ void main() {
...
+
+class _FakeFileStat extends Fake implements FileStat {
+  _FakeFileStat(this.size);
+
+  @override
+  final int size;
+}
+
+class _FakeFile extends Fake implements File {
+  _FakeFile(this.path);
```
Comment (andrewkolos) [inline pkgs/unified_analytics/test/log_handler_test.dart:346]: Out of curiosity, why use mutable fields for method implementations over optional constructor parameters? Do you find this more readable?

File: pkgs/unified_analytics/lib/src/log_handler.dart (hunk 1 above)
Comment (christopherfujino) [reply pkgs/unified_analytics/lib/src/log_handler.dart:284]: This will incur an additional file disk read

**Review by andrewkolos — APPROVED** [review]
LGTM

File: pkgs/unified_analytics/lib/src/log_handler.dart (hunk 1 above)
Comment (andrewkolos) [reply pkgs/unified_analytics/lib/src/log_handler.dart:284]: SGTM. The extra disk read would (_probably_) not happen very often, but I respect being cautious here.

File: pkgs/unified_analytics/pubspec.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ description: >-
   to Google Analytics.
 # When updating this, keep the version consistent with the changelog and the
 # value in lib/src/constants.dart.
-version: 6.1.1
+version: 6.1.2
```
[the remark this answers is not in the document]
Comment (christopherfujino) [reply pkgs/unified_analytics/pubspec.yaml:7]: > or rev to the next approipriate semver version, land the PR, and then use the link in the publishing comment to publish that version.

Ahh, I missed the last step (link in the comment to publish). Thanks.

**Pull request merged**